src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    let nextState = { ...currentState };
+
+    switch (action.type) {
+      case 'clear':
+        nextState = {};
```

Comment:
this also works, but I think we would rather want to `delete` the fields from an object, so as you did in `removeProperties`, we don't really want to reassign empty object, we want to modify the existing one